### PR TITLE
new post  "Exploration and Practice of Serverless Architecture Evolution in Microservices"

### PR DESCRIPTION
**Pull request by betty714**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [x] The title of this pull request reflects the title of my content.
- [x] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [x] This pull request contains one piece of content (post or page).

**Important: if you are submitting a PR to update existing content, please ensure you [sync'ed your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) first before you created your branch.**

Is this pull request linked to a Content Proposal or Idea Suggestion Issue? If so, please link to it below using the number or full link.

```text
Resolves #5

or

Content for https://github.com/build-on-aws/content/issues/5
```

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](https://github.com/build-on-aws/content/blob/main/LICENSE). Notwithstanding the above, if you provide source code text within your content piece, you are providing such source code under the [MIT-0](https://opensource.org/license/mit-0/) license.**

### CODE REVIEW

**Comment by DennisTraub** [comment]
Thank you for your submission.

Question: When finding smaller issues (typos, etc.), do you mind if I directly fix them, or do you prefer to do the fixes yourself?

**Comment by betty714** [comment]
Sure, please directly fix them. 😊


Thanks & Best Regards
Betty Zheng
Developer Advocate, GCR DevRel team



From: Dennis Traub ***@***.***>
Reply-To: build-on-aws/content ***@***.***>
Date: Friday, November 10, 2023 at 15:22
To: build-on-aws/content ***@***.***>
Cc: "Zheng, Betty" ***@***.***>, Assign ***@***.***>
Subject: Re: [build-on-aws/content] new post "Exploration and Practice of Serverless Architecture Evolution in Microservices" (PR #908)


Thank you for your submission.

Question: When finding smaller issues (typos, etc.), do you mind if I directly fix them, or do you prefer to do the fixes yourself?

**Comment by cherbk** [comment]
@betty714 Hi Betty- is this piece still being reviewed by Dennis and Francois?

**Comment by betty714** [comment]
> @betty714 Hi Betty- is this piece still being reviewed by Dennis and Francois?

Hi, @cherbk I think so. They are currently focused on preparing the content for re:Invent and they need an additional week for content review.

File: tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+Elastic Load Balancing (ELB) supports Lambda functions as targets for Application Load Balancers (ALB). With load balancer rules, HTTP requests are routed to a specific function based on the path or header values. The request is processed, and an HTTP response is returned from the Lambda function.
+Setting up the infrastructure requires registering the Lambda function with a Target Group and configuring listener rules in the Application Load Balancer (ALB) to route requests to the Lambda function's Target Group. When the load balancer forwards requests to the Target Group associated with a Lambda function, it triggers the Lambda function and passes the request content in JSON format. This allows the Lambda function to process the request and respond accordingly
+Operating at the application layer, the ALB routes HTTP requests to multiple backend resources, including Amazon Lambda functions. ALBs are commonly used to expose a public web endpoint, with its resources hosted within private subnets in an Amazon VPC; they also support private ALBs, which are only accessible from an organization's internal network.
+![Load banlancer ](./images/4-ALB%2BLambda.png)
```
Comment (DennisTraub) [inline tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md:86]: One target is sitting between two target groups. Does that have any significance?

File: tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+> By choosing the appropriate authorization mechanism and implementing resource-based policies, you can enhance the security of your Lambda Function URL and ensure that only authorized entities have access to it.
+
+## Comparison of Pros and Cons
+Let's briefly compare API gateways, ALBs, and functional URLs in the following ways.
+
+|                                 | API GW                                                       | ALB                                                         | Function URL                                               |
+| ------------------------------- | ------------------------------------------------------------ | ----------------------------------------------------------- | ---------------------------------------------------------- |
+| Scalable                        | Default 10000 RPS                                            | N/A                                                         | 10 * Lambda's reserved concurrency/sec & Default 10000 RPS |
```
Comment (DennisTraub) [inline tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md:111]: Does "N/A" under ALB mean that ALBs are not scalable?

File: tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+| Protocols                       | HTTP/2 (HTTP API, REST API)，WebSockets                     | HTTP/2                                                      | HTTP/2                                                     |
+| Router                          | Path/HTTP Method/IP/HTTP Headers. etc.                       | Path/HTTP Method/IP/HTTP Headers etc,and Configurable load ratios | N/A                             |
+|  Identity and Access Management | Private & IAM and identity-based polic                       | Private & Support for configuring user authentication by creating authentication actions for listener rules | OpenURL & Authentication and authorization using Amazon IAM and resource-based policies. |
+|  Cost                           | Within 3 billion/month: $1.00/million requests & Over3 billion requests/month: $0.90 p/million requests. | 0.0225 USD/hour +（LCU）0.008 USD/hour              | Within 6MB/request: Free & over 6M/request：0.008USD/GB   |
```
Comment (DennisTraub) [inline tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md:116]: "Within 3 billion/month..." 3 billion what?

Also, is mentioning different price tiers even relevant when you only have 100,000 requests per hour?

Comment (DennisTraub) [inline tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md:116]: What is LCU?

File: tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+![pic6](./images/6-Code%20Migration%20Process2.png)
+
+## Step 1: Create a Lambda Function
+1. Open the [Functions page](https://console.aws.amazon.com/lambda/home#/functions) on the Lambda console.
+2. Choose "Create function."
+3. Select "Author from scratch."
+4. Enter the function name: "ad_debug."
+5. For Runtime, select `Go 1.x`
+6. Under Execution Role, choose "Create a new role with basic Lambda permissions."
+After creating an empty Lambda, should to upload and deploy the  package. As mentioned above, the service is based on gRPC, and be keeping the business processing part of the code unchanged. the business processing handler will basically look something like this:
```
Comment (DennisTraub) [inline tutorials/Exploration-and-Practice-of-Serverless-Architecture-Evolution-in-Microservicesindex/index.md:140]: "After creating an empty Lambda, should to upload and deploy the package."

Please check this sentence, I don't understand what it means.

**Review by DennisTraub — CHANGES_REQUESTED** [review]
I was only able to review about 50% of the article in the timeframe I allotted for the task. Will revisit later.

In general, the article touches too many different topics and it is unclear to me what it is about. It also seems to make a lot of assumptions about prior knowledge by the reader, including the specifics of the presented use case.

Please consider reducing the scope of the article, clearly introducing the relevant background, and then covering the technical details.

Additionally, the article needs to be reviewed for style, structure, and  correct usage of AWS terms and service names.

**Comment by DennisTraub** [comment]
Thank you for your contribution and the effort put into this article. After careful consideration, we've decided to go in a different direction at this time. We appreciate your understanding and look forward to collaborating on future projects. Your involvement in our community is greatly valued.

**Pull request closed without merging**